go/vt/mysqlctl/schema.go

```diff
@@ -266,15 +265,60 @@ func ResolveTables(ctx context.Context, mysqld MysqlDaemon, dbName string, table
 	return result, nil
 }
 
-// GetColumns returns the columns of table.
-func (mysqld *Mysqld) GetColumns(ctx context.Context, dbName, table string) ([]*querypb.Field, []string, error) {
-	conn, err := getPoolReconnect(ctx, mysqld.dbaPool)
+const (
+	GetColumnNamesQuery = `SELECT COLUMN_NAME as column_name
+		FROM INFORMATION_SCHEMA.COLUMNS
+		WHERE TABLE_SCHEMA = %s AND TABLE_NAME = '%s'
+		ORDER BY ORDINAL_POSITION`
+	GetFieldsQuery = "SELECT %s FROM %s WHERE 1 != 1"
+)
```

Comment:
I wonder if this is going to behave as we'd like when `GENERATED ALWAYS` (virtual) columns are used as you cannot specify a value on INSERT: https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html
```
mysql> create table gt (col1 int, col2 int, col3 bigint generated always as (col1+col2));
Query OK, 0 rows affected (0.00 sec)

mysql> select * from gt where 1!=1;
Field   1:  `col1`
Catalog:    `def`
Database:   `mydb`
Table:      `gt`
Org_table:  `gt`
Type:       LONG
Collation:  binary (63)
Length:     11
Max_length: 0
Decimals:   0
Flags:      NUM

Field   2:  `col2`
Catalog:    `def`
Database:   `mydb`
Table:      `gt`
Org_table:  `gt`
Type:       LONG
Collation:  binary (63)
Length:     11
Max_length: 0
Decimals:   0
Flags:      NUM

Field   3:  `col3`
Catalog:    `def`
Database:   `mydb`
Table:      `gt`
Org_table:  `gt`
Type:       LONGLONG
Collation:  binary (63)
Length:     20
Max_length: 0
Decimals:   0
Flags:      NUM

0 rows in set (0.00 sec)

mysql> insert into gt values (1, 2, 3);
ERROR 3105 (HY000): The value specified for generated column 'col3' in table 'gt' is not allowed.
```

I feel like we should leverage information_schema for everything or we're going to run into various edge cases over time (e.g. [server side query rewrites](https://dev.mysql.com/doc/refman/8.0/en/rewriter-query-rewrite-plugin.html) is another potential issue):
```
mysql> select * from information_schema.columns where table_name="gt"\G
*************************** 1. row ***************************
           TABLE_CATALOG: def
            TABLE_SCHEMA: mydb
              TABLE_NAME: gt
             COLUMN_NAME: col1
        ORDINAL_POSITION: 1
          COLUMN_DEFAULT: NULL
             IS_NULLABLE: YES
               DATA_TYPE: int
CHARACTER_MAXIMUM_LENGTH: NULL
  CHARACTER_OCTET_LENGTH: NULL
       NUMERIC_PRECISION: 10
           NUMERIC_SCALE: 0
      DATETIME_PRECISION: NULL
      CHARACTER_SET_NAME: NULL
          COLLATION_NAME: NULL
             COLUMN_TYPE: int
              COLUMN_KEY:
                   EXTRA:
              PRIVILEGES: select,insert,update,references
          COLUMN_COMMENT:
   GENERATION_EXPRESSION:
                  SRS_ID: NULL
*************************** 2. row ***************************
           TABLE_CATALOG: def
            TABLE_SCHEMA: mydb
              TABLE_NAME: gt
             COLUMN_NAME: col2
        ORDINAL_POSITION: 2
          COLUMN_DEFAULT: NULL
             IS_NULLABLE: YES
               DATA_TYPE: int
CHARACTER_MAXIMUM_LENGTH: NULL
  CHARACTER_OCTET_LENGTH: NULL
       NUMERIC_PRECISION: 10
           NUMERIC_SCALE: 0
      DATETIME_PRECISION: NULL
      CHARACTER_SET_NAME: NULL
          COLLATION_NAME: NULL
             COLUMN_TYPE: int
              COLUMN_KEY:
                   EXTRA:
              PRIVILEGES: select,insert,update,references
          COLUMN_COMMENT:
   GENERATION_EXPRESSION:
                  SRS_ID: NULL
*************************** 3. row ***************************
           TABLE_CATALOG: def
            TABLE_SCHEMA: mydb
              TABLE_NAME: gt
             COLUMN_NAME: col3
        ORDINAL_POSITION: 3
          COLUMN_DEFAULT: NULL
             IS_NULLABLE: YES
               DATA_TYPE: bigint
CHARACTER_MAXIMUM_LENGTH: NULL
  CHARACTER_OCTET_LENGTH: NULL
       NUMERIC_PRECISION: 19
           NUMERIC_SCALE: 0
      DATETIME_PRECISION: NULL
      CHARACTER_SET_NAME: NULL
          COLLATION_NAME: NULL
             COLUMN_TYPE: bigint
              COLUMN_KEY:
                   EXTRA: VIRTUAL GENERATED
              PRIVILEGES: select,insert,update,references
          COLUMN_COMMENT:
   GENERATION_EXPRESSION: (`col1` + `col2`)
                  SRS_ID: NULL
3 rows in set (0.00 sec)
```

It shouldn't be an issue in replicating binary log events (I don't think), but it might cause problems in the copy rows phase?

Reply:
Generated columns are handled by vreplication, so we drop them from the generated table plans.